process/smartContract/process.go

```diff
@@ -1250,12 +1238,9 @@ func (sc *scProcessor) createSCRsWhenError(
 		if callType == vmcommon.AsynchronousCall {
 			scr.CallType = vmcommon.AsynchronousCallBack
 			scr.GasPrice = tx.GetGasPrice()
-			sc.mutGasLock.RLock()
-			gasToLock := sc.asyncCallStepCost + sc.asyncCallbackGasLock
-			sc.mutGasLock.RUnlock()
-			if tx.GetGasLimit() >= gasToLock {
-				scr.GasLimit = gasToLock
-				consumedFee = core.SafeMul(tx.GetGasPrice(), tx.GetGasLimit()-gasToLock)
+			if tx.GetGasLimit() >= gasLocked {
+				scr.GasLimit = gasLocked
```

Comment:
`scr.GasLocked`? Also add `scr.GasLimit = 0`.

Reply:
there is no scr.GasLocked.